packages/validation/src/nodebalancers.schema.ts

```diff
@@ -44,7 +45,39 @@ export const nodeBalancerConfigNodeSchema = object({
   address: string()
     .typeError('IP address is required.')
     .required('IP address is required.')
-    .matches(PRIVATE_IPV4_REGEX, PRIVATE_IPV4_WARNING),
+    .test(
+      'IP validation',
+      'Must be a private IPv4 or a valid IPv6 address',
+      function (value) {
+        const type = determineIPType(value);
+        const isIPv4 = type === 'ipv4';
+        const isIPv6 = type === 'ipv6';
+
+        if (!isIPv4 && !isIPv6) {
+          return this.createError({
+            message: IPV4_IPV6_WARNING,
+          });
+        }
+
+        if (isIPv4) {
+          if (!PRIVATE_IPV4_REGEX.test(value)) {
+            return this.createError({
+              message: PRIVATE_IPV4_WARNING,
+            });
+          } else {
+            return true;
+          }
+        }
```

Comment:
```suggestion
        if (isIPv4) {
          if (!PRIVATE_IPV4_REGEX.test(value)) {
            return this.createError({
              message: PRIVATE_IPV4_WARNING,
            });
          }
          return true;
        }
```

optional nit: We can remove the unnecessary else block to make it a bit cleaner